cluster/shard_writer.go

```diff
@@ -43,7 +43,9 @@ func (w *ShardWriter) WriteShard(shardID, ownerID uint64, points []tsdb.Point) e
 	if !ok {
 		panic("wrong connection type")
 	}
-	defer conn.Close() // return to pool
+	defer func(conn net.Conn) {
```

Comment:
The connection pool needs to know if a connection becomes `unusable`.  This is very common, and can happen for a number of reasons.  Basically, if we have an error on that connection, we mark it `unusable`, remove it from the pool, and let the pool factory a new one when needed, thus `purging` any potentially bad connections.

The cause we were seeing was when we purposefully dropped a node.  Those connections are in fact bad now as they don't have a connection to the server, so they needed to be removed from the pool, and new ones created when the node came back up.